optimade/server/routers/utils.py

```diff
@@ -274,7 +274,31 @@ def get_entries(
         # Deduce the `next` link from the current request
         query = urllib.parse.parse_qs(request.url.query)
         if isinstance(results, list):
-            query["page_offset"] = int(query.get("page_offset", [0])[0]) + len(results)  # type: ignore[assignment,list-item]
+            # If a user passed a particular pagination mechanism, keep using it
+            # Otherwise, use the default pagination mechanism of the collection
+            for pagination_key in (
+                "page_offset",
+                "page_number",
+                "page_above",
+            ):
+                if query.get(pagination_key, None) is not None:
+                    pagination_mechanism = PaginationMechanism(pagination_key)
+                    break
+            else:
+                pagination_mechanism = collection.pagination_mechanism
+
+            if pagination_mechanism is None:
+                pagination_mechanism = PaginationMechanism.OFFSET
+
+            if pagination_mechanism == PaginationMechanism.OFFSET:
+                query["page_offset"] = [
+                    str(int(query.get("page_offset", [0])[0]) + len(results))  # type: ignore[list-item]
+                ]
+            elif pagination_mechanism == PaginationMechanism.NUMBER:
+                query["page_number"] = [str(int(query.get("page_number", [1])[0]) + 1)]  # type: ignore[list-item]
+            elif pagination_mechanism == PaginationMechanism.ABOVE:
+                query["page_above"] = [str(results[-1].id)]  # type: ignore[union-attr]
```

Comment:
This is a bit too simple. It is only `id` if you sort by id. The `page_above` value is inherently tight to the sorting. Here the sort property and order are important. The sorted-by property defines the values. The order defines if you have to use above vs. below.  If you really want to do it right, you also have to include the id as a tie breaker. 

If you want to keep it simply, you could force sort-by id (via warning/error) for queries with `page_above`.

Reply:
I think that it would be best to handle this case properly here. (or in an appropriate EntriesCollection class method, as suggested in PR#1557.)
Otherwise, each database that wants to support `page_above` will have to reinvent this.